tests/karate/src/test/java/fe/net/ReplyMethods.java

```diff
@@ -17,15 +22,77 @@ public class ReplyMethods {
+      "{\"jsonrpc\":\"2.0\",\"method\": \"broadcast\",\"params\":%PARAM%}";
+
+  private static Json laoCreatePublishJson;
+
+  private static List<String> buildSingleton(String string) {
+    return Collections.singletonList(string);
+  }
+
   /** Always reply with a valid response */
-  public static Function<String, String> ALWAYS_VALID =
+  public static Function<String, List<String>> ALWAYS_VALID_CONSENSUS =
       msg -> {
         Json msgJson = Json.of(msg);
-        int id = msgJson.get("id");
+        int id = msgJson.get(ID);
         String template =
-            msgJson.get("method").equals("catchup")
+            msgJson.get(METHOD).equals(CATCHUP)
                 ? VALID_CATCHUP_REPLY_TEMPLATE
                 : VALID_REPLY_TEMPLATE;
-        return template.replace("%ID%", Integer.toString(id));
+        return buildSingleton(template.replace("%ID%", Integer.toString(id)));
+      };
+
+  /** Always reply with a valid response */
+  public static Function<String, List<String>> ALWAYS_VALID =
+      msg -> {
+        Json msgJson = Json.of(msg);
+        int id = msgJson.get(ID);
+        return buildSingleton(VALID_REPLY_TEMPLATE.replace("%ID%", Integer.toString(id)));
+      };
+
+  public static Function<String, List<String>> LAO_CREATE_CATCHUP =
+      msg -> {
+        if (msg.contains(CONSENSUS) || msg.contains(COIN)) {
+          return ALWAYS_VALID_CONSENSUS.apply(msg);
+        }
+        Json msgJson = Json.of(msg);
+        String replaceId =
+            VALID_CATCHUP_REPLY_TEMPLATE.replace("%ID%", Integer.toString(msgJson.get(ID)));
+        if (laoCreatePublishJson == null) {
+          throw new IllegalStateException(
+              "When creating a catchup the laoCreate should not be null");
+        }
+        return buildSingleton(replaceId.replace("[]", "[" + laoCreatePublishJson + "]"));
+      };
+
+  public static Function<String, List<String>> LAO_CREATE =
+      msg -> {
+        Json msgJson = Json.of(msg);
+        String method = msgJson.get(METHOD);
+        if (PUBLISH.equals(method)) {
+          laoCreatePublishJson = getJSON(getJSON(Json.of(msg), PARAMS), MESSAGE);
+        }
+        if (CATCHUP.equals(method)) {
+          return LAO_CREATE_CATCHUP.apply(msg);
+        } else { // We want to respond valid result for both publish and subscribe
+          return ALWAYS_VALID.apply(msg);
+        }
+      };
+
+  public static Function<String, List<String>> ROLL_CALL_CREATE_BROADCAST =
+      msg -> {
+        Json param = getJSON(Json.of(msg), PARAMS);
+        String channel = param.get(CHANNEL);
+
+        Map<String, String> msgMap = param.get(MESSAGE);
+        Json send = Json.object();
+        send.set(CHANNEL, channel);
+        send.set(MESSAGE, msgMap);
+
+        String broadCast = RC_CREATE_BROADCAST_TEMPLATE.replace("%PARAM%", send.toString());
+        String result = ALWAYS_VALID.apply(msg).get(0);
+
+        return Arrays.asList(broadCast, result);
       };
```

Comment:
I don't see where this code is being used.

Reply:
This is used by the `replyProducer` in the `MockBackend`. To avoid adding complexity to the mock backend, we select manually which reply method we want with the `setXMode`.